service/src/ConstConfigIOGroup.cpp

```diff
@@ -0,0 +1,424 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "ConstConfigIOGroup.hpp"
+
+#include <cmath>
+
+#include <iostream>
+#include <iterator>
+#include <sstream>
+
+#include "geopm/Helper.hpp"
+#include "geopm/PlatformTopo.hpp"
+#include "geopm/Exception.hpp"
+#include "geopm/Agg.hpp"
+#include "geopm_topo.h"
+#include "config.h"
+
+using json11::Json;
+
+namespace geopm
+{
+    const std::string ConstConfigIOGroup::M_PLUGIN_NAME = "CONST_CONFIG";
+    const std::string ConstConfigIOGroup::M_SIGNAL_PREFIX = M_PLUGIN_NAME + "::";
+    const std::map<std::string, ConstConfigIOGroup::m_signal_desc_s>
+        ConstConfigIOGroup::M_SIGNAL_FIELDS = {
+            {"description", {Json::STRING, true}},
+            {"units", {Json::STRING, true}},
+            {"domain", {Json::STRING, true}},
+            {"aggregation", {Json::STRING, true}},
+            {"values", {Json::ARRAY, true}}
+        };
+
+    ConstConfigIOGroup::ConstConfigIOGroup()
+    {
+        // TODO
+    }
+
+    ConstConfigIOGroup::ConstConfigIOGroup(const std::string &config)
+    {
+        parse_config_json(config);
```

Comment:
Where is the checking to see if the config file exists?  As it is, it seems if you pass a string to the parse method, and the file does not exist that this will throw, and construction will fail.

Reply:
Also, where is the file reading code?

Reply:
It's coming :). I first wanted to enable Konstantin to start writing unit tests.